usr/share/rear/output/RAWDISK/Linux-i386/260_create_syslinux_efi_bootloader.sh

```diff
@@ -4,8 +4,11 @@
 
 ### Check prerequisites
 
-# Run only if no EFI bootloader has been created yet and syslinux is available
-([[ -n "$RAWDISK_BOOT_EFI_STAGING_ROOT" ]] || ! has_binary syslinux) && return 0
+# Run only if
+# a) no EFI bootloader has been created yet and
+# b) secure boot (which is assumed to require Grub) is disabled and
+# c) syslinux is available.
+([[ -n "$RAWDISK_BOOT_EFI_STAGING_ROOT" || -n "$SECURE_BOOT_BOOTLOADER" ]] || ! has_binary syslinux) && return 0
```

Comment:
I looked long at this subshell construct and I think it works
but I think it is fragile because the return status of the subshell
is the exit status of the last command executed, e.g.:
```
# ( cat qqq ; cat /etc/issue ) ; echo subshell exit code is $?
cat: qqq: No such file or directory
Welcome to \S - Kernel \r (\l).
subshell exit code is 0

# ( cat /etc/issue ; cat qqq ) ; echo subshell exit code is $?
Welcome to \S - Kernel \r (\l).
cat: qqq: No such file or directory
subshell exit code is 1

# ( cat qqq || cat /etc/issue ) ; echo subshell exit code is $?
cat: qqq: No such file or directory
Welcome to \S - Kernel \r (\l).
subshell exit code is 0

# ( cat /etc/issue || cat qqq ) ; echo subshell exit code is $?
Welcome to \S - Kernel \r (\l).
subshell exit code is 0
```
plus the De Morgan's law transformation of the condition in the comment
makes it hard for me to see that this code implements the condition in the comment.
Why not implement the skip conditions separatedly and directly?
```
# skip if EFI bootloader has been created 
test "$RAWDISK_BOOT_EFI_STAGING_ROOT" && return
# skip if secure boot (which is assumed to require Grub) is enabled
test "$SECURE_BOOT_BOOTLOADER" && return
# skip if syslinux is not available
has_binary syslinux || return 0
```